src/main.rs

```diff
@@ -141,10 +141,12 @@ pub enum ShredstreamProxyError {
 }
 
 fn resolve_hostname_port(hostname_port: &str) -> io::Result<(SocketAddr, String)> {
-    let socketaddr = hostname_port.to_socket_addrs()?.next().ok_or(Error::new(
-        ErrorKind::AddrNotAvailable,
-        format!("Could not find destination {hostname_port}"),
-    ))?;
+    let socketaddr = hostname_port.to_socket_addrs()?.next().ok_or_else(|| {
```

Comment:
curious what is the benefit of using `ok_or_else`?

Reply:
https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or

> Arguments passed to ok_or are eagerly evaluated; if you are passing the result of a function call, it is recommended to use [ok_or_else](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else), which is lazily evaluated.